hazelcast/src/main/java/com/hazelcast/jet/pipeline/SinkBuilder.java

```diff
@@ -167,6 +170,21 @@ private SinkBuilder(
         return this;
     }
 
+    /**
+     * Sets the the permission required to use this sink when the
+     * security is enabled. The default value is {@code null} which
+     * means there is no restriction to use this sink. Security is an
+     * enterprise feature.
+     *
+     * @param permission the required permission to use this sink when
+     *     security is enabled.
+     */
+    public SinkBuilder<C, T> permission(@Nonnull Permission permission) {
```

Comment:
We have `permission` in the generic `SinkBuilder`, but not in `SourceBuilder`. Omission? 
You can't do much with these builders without custom code upload, and with custom code upload the security is pointless. But there is one use case: the admin might add some contrib connectors to the member classpath and for those, this permission might be useful.

Reply:
adding permission to the sink builder was necessary because we create ring-buffer sink using the sink-builder. I can add it to source-builder too for completeness.